daemon/api_notices.go

```diff
@@ -76,7 +81,7 @@ func getNotices(c *Command, r *http.Request, user *auth.UserState) Response {
 		userID = nil
 	}
 
-	types, err := sanitizeTypesFilter(query["types"])
+	types, err := sanitizeNoticeTypesFilter(query["types"])
```

Comment:
I went through the user filtering code, it seems that specifying the user filter seems to give permission denied instead of filtering out users's notices:
https://github.com/snapcore/snapd/blob/b280382184dc840f93f2bfefc666cf7c2e72431e/daemon/api_notices.go#L56-L58
https://github.com/snapcore/snapd/blob/b280382184dc840f93f2bfefc666cf7c2e72431e/daemon/api_notices.go#L66-L68
https://github.com/snapcore/snapd/blob/b280382184dc840f93f2bfefc666cf7c2e72431e/daemon/api_notices.go#L195-L197

which I think is the right approach if the user is intentionally trying to see other user's notices to get denied (the internal filtering afterwards makes sense). I think it is a similar case for snaps trying to access notices, We are effectively decreasing the field of view of the snap to certain notice types and forcing it to use it this way explicitly. If we want generic access we can easily create a `notices-observer` interface, but then the snaps would need to explicitly request this interface instead of trying the api and getting no notices in response if they choose the wrong subset of notice types.

Reply:
Yes, what I mean is that by default (no `user-id` or `users` parameters), it returns just the notices for the UID which made the request -- that is, the ones they're allowed to view.

https://github.com/snapcore/snapd/blob/b280382184dc840f93f2bfefc666cf7c2e72431e/daemon/api_notices.go#L52-L77

There isn't a need to pass one's own userID, it's implied by default. A client can pass another userID explicitly to use instead, but only if they're root. And they can remove the implicit userID filter and get notices for all users if they pass `users=all`.

That's what I'm interested in here: constructing the types filter in the API to by default filter on the types which are allowed by the interface. And if a type(s) are explicitly included, those override the default types filter. If a type is listed which is not allowed by the interface, an error is returned.

The motivation for this would be so that snap clients can just query the notices endpoint without a filter and get the notices they're allowed to view (according to connected interfaces). But perhaps that's not motivation enough?